### PR TITLE
Add note about not monkey-patching builtins

### PR DESCRIPTION
**Pull request by nicoddemus**
Related to #1985

### CODE REVIEW

File: doc/en/monkeypatch.rst [hunk 1]
```diff
@@ -55,6 +55,14 @@ will delete the method ``request.session.Session.request``
 so that any attempts within tests to create http requests will fail.
 
 
+.. note::
+    
+    Be advised that it is not recommended to patch builtin functions such as ``open``,
+    ``compile``, etc., because it might break pytest's internals. If that's
+    unavoidable, passing ``--tb=native`` and/or ``--assert=plain`` might help althought
```
Comment (RonnyPfannschmidt) [inline doc/en/monkeypatch.rst]: also `-s` ?

Comment (nicoddemus) [inline doc/en/monkeypatch.rst]: Good point!

**Comment by nicoddemus** [comment]
(once this passes let's squash and merge please)

**Comment by RonnyPfannschmidt** [comment]
only trial fails

**Comment by nicoddemus** [comment]
Thanks!

**Pull request merged**